incubator/hnc/internal/reconcilers/object.go

```diff
@@ -431,7 +432,7 @@ func (r *ObjectReconciler) syncSource(ctx context.Context, log logr.Logger, src
 	ns := r.Forest.Get(src.GetNamespace())
 	ns.SetSourceObject(src.DeepCopy())
 
-	if !r.shouldPropagateSource(log, src) {
+	if !r.shouldPropagateSource(log, src, nil) {
```

Comment:
Add a comment here that it only checks if the source itself is propagatable in general but not to a specific destination. We will enqueue all the descendants if it's propagatable and decide whether to propagate when reconciling the descendants.

Reply:
done